examples/keras_spark_rossmann.py

```diff
@@ -0,0 +1,554 @@
+        hvd.callbacks.MetricAverageCallback(),
+
+        # Horovod: using `lr = 1.0 * hvd.size()` from the very beginning leads to worse final
+        # accuracy. Scale the learning rate `lr = 1.0` ---> `lr = 1.0 * hvd.size()` during
+        # the first five epochs. See https://arxiv.org/abs/1706.02677 for details.
+        hvd.callbacks.LearningRateWarmupCallback(warmup_epochs=5, verbose=verbose),
+
+        # Reduce LR if the metric is not improved for 10 epochs, and stop training
+        # if it has not improved for 20 epochs.
+        tf.keras.callbacks.ReduceLROnPlateau(monitor='val_exp_rmspe', patience=10, verbose=verbose),
+        tf.keras.callbacks.EarlyStopping(monitor='val_exp_rmspe', mode='min', patience=20, verbose=verbose),
+        tf.keras.callbacks.TerminateOnNaN()
+    ]
+
+    # Model checkpoint location.
+    ckpt_dir = tempfile.mkdtemp()
+    ckpt_file = os.path.join(ckpt_dir, 'checkpoint.h5')
+    atexit.register(lambda: shutil.rmtree(ckpt_dir))
+
+    # Horovod: save checkpoints only on the first worker to prevent other workers from corrupting them.
+    if hvd.rank() == 0:
+        callbacks.append(tf.keras.callbacks.ModelCheckpoint(ckpt_file, monitor='val_exp_rmspe', mode='min',
+                                                            save_best_only=True))
+
+    # Make Petastorm readers.
+    with make_batch_reader('%s/train_df.parquet' % DATA_LOCATION, num_epochs=None,
+                           cur_shard=hvd.rank(), shard_count=hvd.size(),
+                           hdfs_driver=PETASTORM_HDFS_DRIVER) as train_reader:
+        with make_batch_reader('%s/val_df.parquet' % DATA_LOCATION, num_epochs=None,
+                               cur_shard=hvd.rank(), shard_count=hvd.size(),
+                               hdfs_driver=PETASTORM_HDFS_DRIVER) as val_reader:
+            # Convert readers to tf.data.Dataset.
+            train_ds = make_petastorm_dataset(train_reader) \
+                .apply(tf.data.experimental.unbatch()) \
+                .shuffle(int(train_rows / hvd.size())) \
+                .batch(BATCH_SIZE) \
+                .map(lambda x: (tuple(getattr(x, col) for col in all_cols), tf.log(x.Sales)))
+
+            val_ds = make_petastorm_dataset(val_reader) \
+                .apply(tf.data.experimental.unbatch()) \
+                .batch(BATCH_SIZE) \
+                .map(lambda x: (tuple(getattr(x, col) for col in all_cols), tf.log(x.Sales)))
+
+            history = model.fit(train_ds,
+                                validation_data=val_ds,
+                                steps_per_epoch=int(train_rows / BATCH_SIZE / hvd.size()),
+                                validation_steps=int(val_rows / BATCH_SIZE / hvd.size()),
+                                callbacks=callbacks,
+                                verbose=verbose,
+                                epochs=100)
+
+    # TODO (@yevgeni): Petastorm currently displays a wall of errors upon termination.
```

Comment:
@selitvin, this is the current hack that I have to avoid Petastorm errors upon process exit.